test/e2e/kubectl.go

```diff
@@ -1140,23 +1140,37 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 				framework.Failf("Failed creating a job with correct restart policy for --restart=OnFailure")
 			}
 		})
+	})
+
+	framework.KubeDescribe("Kubectl run pod", func() {
+		var nsFlag string
+		var podName string
+
+		BeforeEach(func() {
+			nsFlag = fmt.Sprintf("--namespace=%v", ns)
+			podName = "e2e-test-nginx-pod"
+		})
+
+		AfterEach(func() {
+			framework.RunKubectlOrDie("delete", "pods", podName, nsFlag)
+		})
 
-		It("should create a job from an image when restart is Never [Conformance]", func() {
+		It("should create a pod from an image when restart is Never [Conformance]", func() {
 			framework.SkipUnlessServerVersionGTE(jobsVersion, c)
 
 			By("running the image " + nginxImage)
-			framework.RunKubectlOrDie("run", jobName, "--restart=Never", "--image="+nginxImage, nsFlag)
-			By("verifying the job " + jobName + " was created")
-			job, err := c.Extensions().Jobs(ns).Get(jobName)
+			framework.RunKubectlOrDie("run", podName, "--restart=Never", "--image="+nginxImage, nsFlag)
```

Comment:
We should add `--generator=xxx` here and in the jobs section so that we make sure we're testing the same thing. To test the default behavior (i.e. `Never` creates pods) it should sits under "kubectl run default".